msticpy/transform/iocextract.py

```diff
@@ -125,18 +127,35 @@ class IoCExtract:
     """
 
     IPV4_REGEX = r"(?P<ipaddress>(?:[0-9]{1,3}\.){3}[0-9]{1,3})"
+    IPV4_DF_REGEX = r"(?P<ipaddress>(?:[0-9]{1,3}\[?\.\]?){3}[0-9]{1,3})"
     IPV6_REGEX = r"(?<![:.\w])(?:[A-F0-9]{0,4}:){2,7}[A-F0-9]{0,4}(?![:.\w])"
     DNS_REGEX = r"((?=[a-z0-9-]{1,63}\.)[a-z0-9]+(-[a-z0-9]+)*\.){1,126}[a-z]{2,63}"
+    DNS_DF_REGEX = (
+        r"((?=[a-z0-9-]{1,63}\[?\.\]?)[a-z0-9]+(-[a-z0-9]+)*\[?\.\]?){1,126}[a-z]{2,63}"
+    )
+
+    EMAIL_USER_REGEX = r"(?P<user>[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+)(@|AT)"
+    EMAIL_REGEX = f"{EMAIL_USER_REGEX}(?P<domain>{DNS_REGEX})"
+    EMAIL_DF_REGEX = f"{EMAIL_USER_REGEX}(?P<domain>{DNS_DF_REGEX})"
 
     URL_REGEX = r"""
-            (?P<protocol>(https?|ftp|telnet|ldap|file)://)
+            (?P<protocol>(https?|s?ftps?|telnet|ldap|file)://)
             (?P<userinfo>([a-z0-9-._~!$&\'()*+,;=:]|%[0-9A-F]{2})*@)?
             (?P<host>([a-z0-9-._~!$&\'()*+,;=]|%[0-9A-F]{2})*)
             (:(?P<port>\d*))?
             (/(?P<path>([^?\#"<>\s]|%[0-9A-F]{2})*/?))?
             (\?(?P<query>([a-z0-9-._~!$&'()*+,;=:/?@]|%[0-9A-F]{2})*))?
             (\#(?P<fragment>([a-z0-9-._~!$&'()*+,;=:/?@]|%[0-9A-F]{2})*))?"""
 
+    URL_DF_REGEX = r"""
```

Comment:
Weird - we have a bunch of very similar code (in the defanging PR). Nice!

Reply:
Or is some of this merged from that PR?

Reply:
Yeah I merged some stuff for testing the notebook with them, I will merge the rest.